File: csharp/Facade/CustomerManager.cs
namespace Facade;

public class CustomerManager
{
    private CrossCuttingConcernsFacade _crossCuttingConcernsFacade;

    public CustomerManager()
    {
        _crossCuttingConcernsFacade = new CrossCuttingConcernsFacade();
    }

    public void Save()
    {
        _crossCuttingConcernsFacade.Caching.Cache();
        _crossCuttingConcernsFacade.Logging.Log();
        Console.WriteLine("Saved");
    }
}

File: java/facade/src/main/CustomerManager.java
package main;

public class CustomerManager {
    public CrossCuttingConcerns crossCuttingConcerns;

    public CustomerManager()
    {
        crossCuttingConcerns = new CrossCuttingConcerns();
    }

    public void save()
    {
        crossCuttingConcerns.caching.cache();
        crossCuttingConcerns.logging.log();
        System.out.println("saved");
    }
}
